### PR TITLE
feat: add functionallity to update Github status through Http Request

### LINKED ISSUE #12: Use HTTP request to update Github status
Depending on the results, update the status of the commit on Github

Good links:
* https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks
* https://docs.github.com/en/rest/reference/commits#commit-statuses

### PR DESCRIPTION
**Pull request by SimonOsnes**
fix #12

### CODE REVIEW

File: src/main/java/group8/git/GitCommitStatus.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            httpPost.addHeader("Authorization", "token " + Config.GitToken);
+
+            StringEntity se = new StringEntity(jsonObject.toString());
+
+            httpPost.setEntity(se);
+            CloseableHttpResponse response = httpClient.execute(httpPost);
+
+            // TODO Remove
+            System.out.println(response.getStatusLine());
+        }
```
Comment (emilmyresten) [inline src/main/java/group8/git/GitCommitStatus.java:75]: Should add a finally { httpClient.close() } here to free the resources used by the httpclient, see https://stackoverflow.com/questions/21574478/what-is-the-difference-between-closeablehttpclient-and-httpclient-in-apache-http

**Review by emilmyresten — COMMENTED** [review]
Commented a change that should be added, otherwise looks great!
It obviously doesn't pass the github actions tests either since github doesnt have access to config class, perhaps an API key can be added as an environment variable to solve this?

**Comment by SimonOsnes** [comment]
Added the Config.java with empty GitToken variable. So for anyone who want to test changing the status should add their own Personal Access token but **NOT** push it up to Github under any circumstances.

As for the closing of the HTTP client, I can add the close() right away

File: src/main/java/group8/git/GitCommitStatus.java (hunk 1 above)
Comment (SimonOsnes) [reply src/main/java/group8/git/GitCommitStatus.java:75]: image

Oh I guess not then

Comment (emilmyresten) [reply src/main/java/group8/git/GitCommitStatus.java:75]: Checked it out apparently try-catch manages resources on its own, so only needed if the httpClient is initiated before the try-catch! :) Everything looks good to me now!

**Review by emilmyresten — APPROVED** [review]
LGTM!

**Pull request merged**